### PR TITLE
Complete the `instantiate_internal` function

### PR DESCRIPTION
**Pull request by yzhang90**
Complete #653

### CODE REVIEW

File: cairo/src/verifier.cairo [hunk 1]
```diff
@@ -348,21 +348,61 @@ fn instantiate_internal(
...
+            }
+        },
+        Pattern::Exists(ExistsType{var,
+        subpattern }) => {
+            let mut subpattern = subpattern.unwrap().unbox();
+
+            let mut new_sub = instantiate_internal(ref subpattern, ref vars, ref plugs);
+
+            if new_sub.is_none() {
+                Option::None
```
Comment (Robertorosmaninho) [inline cairo/src/verifier.cairo:378]: Cairo must return None here?

File: cairo/src/verifier.cairo [hunk 2]
```diff
@@ -348,21 +348,61 @@ fn instantiate_internal(
...
+            }
+        },
+        Pattern::Mu(MuType{var,
+        subpattern }) => {
+            let mut subpattern = subpattern.unwrap().unbox();
+
+            let mut new_sub = instantiate_internal(ref subpattern, ref vars, ref plugs);
+
+            if new_sub.is_none() {
+                Option::None
```
Comment (Robertorosmaninho) [inline cairo/src/verifier.cairo:390]: Same question here.

**Review by Robertorosmaninho — APPROVED** [review]
Approved with some questions

File: cairo/src/verifier.cairo (hunk 1 above)
Comment (yzhang90) [reply cairo/src/verifier.cairo:378]: The reference impl is
```
Some(exists(*var, new_sub?))
```
That  ? means if new_sub is none, then the whole expression is None.

**Pull request merged**